lib/contract.ts

```diff
@@ -1032,16 +1032,102 @@ export default class Contract {
 			[] as Contract[],
 		);
 	}
+
+	private isRequirementSatisfied(
+		requirement: Contract,
+		options: { types?: Set<string> } = {},
+	): boolean {
+		// Utilities
+		const shouldEvaluateType = (type: string) =>
+			options.types ? options.types.has(type) : true;
+
+		/**
+		 * @summary Check if a matcher is satisfied
+		 * @function
+		 * @public
+		 *
+		 * @param {Object} matcher - matcher contract
+		 * @returns {Boolean} whether the matcher is satisfied
+		 *
+		 * @example
+		 * const matcher = Contract.createMatcher({
+		 *   type: 'sw.os',
+		 *   slug: 'debian'
+		 * })
+		 *
+		 * if (hasMatch(matcher)) {
+		 *   console.log('This matcher is satisfied!')
+		 * }
+		 */
+		const hasMatch = (matcher: Contract): boolean => {
+			// TODO: Write a function similar to findContracts
+			// that stops as soon as it finds one match
+			return (
+				this.findChildren(matcher).length > 0 ||
+				this.findChildrenWithCapabilities(matcher).length > 0
+			);
+		};
+
+		if (requirement.raw.operation === 'or') {
+			// (3.1) Note that we should only consider disjuncts
+			// of types we are allowed to check. We can make
+			// such transformation here, so we can then consider
+			// the disjunction as fulfilled if there are no
+			// remaining disjuncts.
+			const disjuncts = filter(requirement.raw.data.getAll(), (disjunct) => {
+				return shouldEvaluateType(disjunct.raw.data.type);
+			});
+			// (3.2) An empty disjuction means that this particular
+			// requirement is fulfilled, so we can carry on.
+			// A disjunction naturally contains a list of further
+			// requirements we need to check for. If at least one
+			// of the members is fulfilled, we can proceed with
+			// next requirement.
+			if (disjuncts.length === 0 || some(map(disjuncts, hasMatch))) {
+				return true;
+			}
+			// (3.3) If no members were fulfilled, then we know
+			// that this requirement was not fullfilled, so it will be returned
+			return false;
+		} else if (requirement.raw.operation === 'not') {
+			// (3.4) Note that we should only consider disjuncts
+			// of types we are allowed to check. We can make
+			// such transformation here, so we can then consider
+			// the disjunction as fulfilled if there are no
+			// remaining disjuncts.
+			const disjuncts = filter(requirement.raw.data.getAll(), (disjunct) => {
+				return shouldEvaluateType(disjunct.raw.data.type);
+			});
+			// (3.5) We fail the requirement if the set of negated
+			// disjuncts is not empty, and we have at least one of
+			// them in the context.
+			if (disjuncts.length > 0 && some(map(disjuncts, hasMatch))) {
```

Comment:
I expect the same here:
```suggestion
			if (disjuncts.length > 0 && some(disjuncts, hasMatch)) {
```

Reply:
Will make the change in #80